siliconcompiler/flows/asicflow.py

```diff
@@ -122,18 +125,19 @@ def setup(chip, flowname='asicflow'):
 
     flowtools = setup_frontend(chip)
     for step in flowpipe:
-        flowtools.append((step, tools[step]))
+        flowtools.append((step, tools[step][0], tools[step][1]))
 
     # Programatically build linear portion of flowgraph and fanin/fanout args
-    for step, tool in flowtools:
+    for step,tool,task in flowtools:
+        #TODO: Fix
         param = step + "_np"
         fanout = 1
         if param in chip.getkeys('arg', 'flow'):
             fanout = int(chip.get('arg', 'flow', param)[0])
         # create nodes
         for index in range(fanout):
             # nodes
-            chip.node(flowname, step, tool, index=index)
+            chip.node(flowname, step, tool, task, index=index)
             # edges
             if re.search(r'join|maximum|minimum|verify', tool):
```

Comment:
task?

Reply:
The node() method was changed to add an optional task argument, letting us handle the new case where multiple steps exist for one task. The default behavior is for step==task.

Reply:
I was actually referring to line 142, but that would be in conjunction with my comments about the builtin functions changing their tool name and taking a task.